packages/contract-notifier/src/ContractNotifier.js

```diff
@@ -0,0 +1,119 @@
+// This notifier is used to monitor a API endpoint serving known contract addresses and their expirations.
+
+const { createEtherscanLinkMarkdown } = require("@uma/common");
+const { Datastore } = require("@google-cloud/datastore");
+const datastore = new Datastore();
+
+class ContractNotifier {
+  /**
+   * @param {Object} logger Winston module used to send logs.
+   * @param {Object} networker Used to send the API requests.
+   * @param {Function} getTime Returns the current time.
+   * @param {String} apiEndpoint API endpoint to monitor.
+   * @param {Integer} maxTimeTillExpiration Period in seconds to look for upcoming contract expirations.
+   */
+  constructor({ logger, networker, getTime, apiEndpoint, maxTimeTillExpiration }) {
+    this.logger = logger;
+    this.networker = networker;
+    this.getTime = getTime;
+    this.apiEndpoint = apiEndpoint;
+    this.maxTimeTillExpiration = maxTimeTillExpiration;
+  }
+
+  async checkUpcomingExpirations() {
+    this.logger.debug({
+      at: "ExpirationsNotifier",
+      message: "Checking for upcoming expirations",
+      apiEndpoint: this.apiEndpoint,
+      maxTimeTillExpiration: this.maxTimeTillExpiration,
+    });
+
+    const currentTime = await this.getTime();
+    const notifiedExpirations = await this.getNotifiedExpirations();
+    const expirationPeriod =
+      this.maxTimeTillExpiration > 259200
+        ? parseInt(this.maxTimeTillExpiration / 86400) + " days"
+        : parseInt(this.maxTimeTillExpiration / 3600) + " hours";
+    const apiUrl = this.apiEndpoint + "/global/listActive";
+    const activeContracts = await this.networker.getJson(apiUrl, { method: "post" });
+    const expiringContracts = activeContracts
+      .map((contract) => {
+        if (!contract.type || !contract.expirationTimestamp) {
+          return null;
+        }
+        let tokenName;
+        if (contract.type === "emp") {
+          tokenName = contract.tokenName;
+        } else if (contract.type === "lsp") {
+          tokenName = contract.longTokenName;
+        } else {
+          tokenName = "";
+        }
+        const expirationUtcString = new Date(contract.expirationTimestamp * 1000).toUTCString();
+        return {
+          // UMA API currently supports only Ethereum mainnet, thus chainId of 1 is hardcoded here:
+          chainId: 1,
```

Comment:
seems fine for now. I know we'll have other chains in the API very soon. when that comes you can sample pass it down from an env.